Feature/sdl 0189 restructuring on reset timeout

LitvinenkoIra opened the pull request:
Just for internal review

This PR is **[ready]** for review.

Need to test

vlrm10 commented on app/controller/sdl/RController.js:
```diff
@@ -660,6 +660,19 @@ SDL.RController = SDL.SDLController.extend(
+    /**
+     * isEmptyObject function. Checks that the object is empty
+     */
+    isEmptyObject: function(object) {
+      var l = 0;
+      for (var key in object) {
+           if(object.hasOwnProperty(key)) {
+               ++l;
+           }
+      }
+      return l == 0;
+   },
```
Better to use :
`return !!Object.keys(object).length`

vlrm10 commented on app/controller/sdl/Abstract/Controller.js:
```diff
@@ -691,7 +686,6 @@ SDL.SDLController = Em.Object.extend(
               SDL.AlertPopUp.deactivate();
             }, SDL.AlertPopUp.timeout
           );
-          this.onResetTimeout(element.appID, 'UI.Alert');
```
`UI.Alert` move to const

vlrm10 replied:
`UI.Alert` move to const

vlrm10 commented on app/controller/sdl/KeyboardController.js:
```diff
@@ -92,9 +92,9 @@ SDL.KeyboardController = Em.Object.create({
-            this.appModel.appID, 'UI.PerformInteraction'
-          );
+          FFW.BasicCommunication.OnResetTimeout(
+            this.requestID, 'UI.PerformInteraction'
```
`UI.PerformInteraction` move to const

vlrm10 commented on app/controller/sdl/KeyboardController.js:
```diff
@@ -148,9 +148,9 @@ SDL.KeyboardController = Em.Object.create({
-            this.appModel.appID, 'UI.PerformInteraction'
-          );
+          FFW.BasicCommunication.OnResetTimeout(
+            this.requestID, 'UI.PerformInteraction'
```
same as above

vlrm10 commented:
There are many hardcoded strings in PR. Should be moved to const

vlrm10 commented on app/view/sdl/ResetTimeoutPopUp.js:
```diff
@@ -0,0 +1,341 @@
+    {
+        self = this;
+        this.resetTimeoutRPCs.forEach(function (method) {
+            if (self.timeoutSeconds[method] == undefined) {
```
Better to use `in` operator to check that the key is defined. Also check all of them in PR

vlrm10 commented on app/view/sdl/ResetTimeoutPopUp.js:
```diff
@@ -0,0 +1,341 @@
+        this.resetTimeOutLabel();
+        clearInterval(this.timer);
+        
+        const self = this;
+        this.timer = setInterval(
+            function() {
+                    var message = '';
+                    self.resetTimeoutRPCs.forEach(function (method) {
+                        self.timeoutSeconds[method] = self.timeoutSeconds[method] - 1;
+                        message = message + method + ": " + self.timeoutSeconds[method].toString() + '\n';
+                     });
+                    self.set('timeoutString',message);
+            }, 1000
+        ); 
```
Better to use arrow function in this case

Commit:
SLD-0189 Restructuring OnResetTimeout
 - Basic implementation of ResetTimeoutPopUp

Commit:
Moved isEmptyObjectMethod to RCOntroller

Commit:
SLD-0189 Restructuring OnResetTimeout

 - Updated ALertManeuverPopUp model
 - Updated AlertPopUp model
 - Updated SubtleAlertPopUp model
 - Updated TTSPopUp model

Commit:
SLD-0189 Restructuring OnResetTimeout

Updated RPC models

Commit:
SLD-0189 Restructuring OnResetTimeout

Updated InteractionChoicesView Models for PI requests

Commit:
SLD-0189 Restructuring OnResetTimeout

Added reset timeout input field

Commit:
SLD-0189 Restructuring OnResetTimeout

 - Updated ScrollableMessage Model
 - Updated SliderView model

Commit:
SLD-0189 Restructuring OnResetTimeout

Minor fixes

LitvinenkoIra commented:
Moved to open
